src/main/java/org/opensearch/knn/quantization/models/quantizationState/QuantizationStateCache.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.quantization.models.quantizationState;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+/**
+ * A thread-safe singleton cache that contains quantization states.
+ */
+public class QuantizationStateCache {
```

Comment:
Questions as this cache will be storing information which is at a segment level:
1. How we are controlling the max size of the cache? and given that this cache will be present in Heap, will the size of the cache be static? How we are controlling the Cache thrashing use case and how as a Opensearch user I will know there is cache thrashing?
2. During eviction, what is the eviction policy getting used for this cache?
3. What controls we have to increase and decrease the size of cache?
4. What controls we are providing to clear the cache do we need it?
5. Should we also have a TTL based eviction?
6. Should we integrate stats of this cache in knn stats api?